elements/editor-viewport/editor-viewport.js

```diff
@@ -63,13 +65,44 @@ Polymer
     this.create_grid(10);
 
     var geometry = new THREE.BoxGeometry(5.0, 5.0, 5.0);
-    var material = new THREE.MeshBasicMaterial({ color: 0xee3987 });
+    var material = new THREE.MeshPhongMaterial({
+                   ambient  : 0x444444,
+                   color    : 0xee3987,
+                   shininess: 300,
+                   specular : 0x33AA33,
+                   shading  : THREE.SmoothShading
+    });
+    geometry.receiveShadow = true;
+    geometry.castShadow = true;
     this.mesh.geometry = geometry;
     this.mesh.material = material;
 
     this.scene.add(this.mesh);
   },
 
+  init_lighting: function()
+  {
+    var ambient = new THREE.AmbientLight(0x666666);
+    this.scene.add(ambient);
+    var direct = new THREE.DirectionalLight(0xdfebff, 1.75);
+    direct.position.set(0,10,10);
+    direct.castShadow = true;
+    direct.shadowCameraVisible = true;
+
+    direct.shadowMapWidth = direct.shadowMapHeight = 2048;
+
+    var d = 50;
```

Comment:
It seems wrong to call variable with a char. I think it would be much better to call i t something like `var direction` or wathever the context is here.